pkg/util/util.go

```diff
@@ -178,7 +178,7 @@ func ParseComponentImageName(imageName string) (string, string, string, string)
 	componentVersion := "latest"
 
 	// Check if componentType includes ":", if so, then we need to spit it into using versions
-	if strings.ContainsAny(componentImageName, ":") {
+	if strings.ContainsAny(componentType, ":") {
```

Comment:
This should remain `componentImageName`

Reply:
Not denying this. I refrained from commenting about this because I was confused.

1. In the function that this line is a part of, both `componentImageName` and `componentType` carry same value as `imageName` without an explanation as to why two variables with same value. So unless I'm mistaken, the `ContainsAny` function should return same value irrespective of the variable name being used.
1. The places where this function is being referenced, we're using `componentType` or the equivalent at a few places which adds to the confusion. See below references from the current master branch 4bb02182db17c445ab5762d97486a7ca5aee886c:
    1. https://github.com/openshift/odo/blob/4bb02182db17c445ab5762d97486a7ca5aee886c/pkg/catalog/catalog.go#L448
    1. https://github.com/openshift/odo/blob/4bb02182db17c445ab5762d97486a7ca5aee886c/pkg/component/component.go#L513
    1. https://github.com/openshift/odo/blob/4bb02182db17c445ab5762d97486a7ca5aee886c/pkg/devfile/adapters/docker/utils/utils.go#L180

Reply:
@kadel I changed it to `componentType` because comment and the conditional check were really confusing for me https://github.com/openshift/odo/pull/4070/files/988fd4cad316cd26157fe5999ac4b144ab786a07#diff-6c6a7fefe37261e2fa8a010cf9e7ae08L180-L181 . I am not sure how `componentImageName` and `componentType` makes the difference. Is the comment line stating `componentType` is misinterpreted or `if strings.ContainsAny(componentImageName, ":")`